src/components/profile-page/access-tokens/access-token-creation-form.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { ChangeEvent, FormEvent } from 'react'
+import React, { Fragment, useCallback, useEffect, useMemo, useState } from 'react'
+import { Button, Form } from 'react-bootstrap'
+import { cypressId } from '../../../utils/cypress-attribute'
+import { Trans, useTranslation } from 'react-i18next'
+import { AccessTokenCreatedModal } from './access-token-created-modal'
+import { DateTime } from 'luxon'
+import type { AccessTokenWithSecret } from '../../../api/tokens/types'
+import { postNewAccessToken } from '../../../api/tokens'
+import { showErrorNotification } from '../../../redux/ui-notifications/methods'
+
+/**
+ * Form for creating a new access token.
+ */
+export const AccessTokenCreationForm: React.FC = () => {
+  const { t } = useTranslation()
+  const [newTokenLabel, setNewTokenLabel] = useState('')
+  const [newTokenExpiry, setNewTokenExpiry] = useState('')
+  const [newTokenWithSecret, setNewTokenWithSecret] = useState<AccessTokenWithSecret>()
+  const [showCreatedModal, setShowCreatedModal] = useState(false)
+
+  const onHideCratedModal = useCallback(() => {
+    setShowCreatedModal(false)
+  }, [])
+
+  const onChangeTokenLabel = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setNewTokenLabel(event.target.value)
+  }, [])
+
+  const onChangeTokenExpiry = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setNewTokenExpiry(event.target.value)
+  }, [])
+
+  const onCreateToken = useCallback(
+    (event: FormEvent<HTMLFormElement>) => {
+      event.preventDefault()
+      postNewAccessToken(newTokenLabel, newTokenExpiry)
+        .then((tokenWithSecret) => {
+          setNewTokenWithSecret(tokenWithSecret)
+          setShowCreatedModal(true)
```

Comment:
Please don't use multiple useStates for states that are tied together! You want to show the token if you have one. Otherwise the modal shouldn't be visible. Every state change triggers a re-render immediately. State are updates are NOT bulked.